src/app/utils/field-validation/validators/textValidator.ts

```diff
@@ -0,0 +1,76 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { pipe } from 'fp-ts/lib/pipeable'
+
+import { ILongTextField, IShortTextField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+import { ISingleAnswerResponse } from 'src/types/response'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type textFieldValidatorConstructor = (
+  textField: IShortTextField | ILongTextField,
+) => ResponseValidator<ISingleAnswerResponse>
+
+const minLengthValidator: textFieldValidatorConstructor = (textField) => (
+  response,
+) => {
+  const { customMin } = textField.ValidationOptions
+  const min = customMin !== null ? Number(customMin) : null
+  if (min === null) return right(response)
+  return response.answer.length >= min
+    ? right(response)
+    : left(`TextValidator.minLength`)
+}
+
+const maxLengthValidator: textFieldValidatorConstructor = (textField) => (
+  response,
+) => {
+  const { customMax } = textField.ValidationOptions
+  const max = customMax !== null ? Number(customMax) : null
+  if (max === null) return right(response)
+  return response.answer.length <= max
+    ? right(response)
+    : left(`TextValidator.maxLength`)
+}
+
+const exactLengthValidator: textFieldValidatorConstructor = (textField) => (
+  response,
+) => {
+  const { customMin, customMax } = textField.ValidationOptions
+  const exact =
+    customMin !== null
+      ? Number(customMin)
+      : customMax !== null
+      ? Number(customMax)
+      : null
+  if (exact === null) return right(response)
+  return response.answer.length === exact
+    ? right(response)
+    : left(`TextValidator.exactLength`)
+}
+
+const lengthValidator: textFieldValidatorConstructor = (textField) => (
+  response,
+) => {
+  switch (textField.ValidationOptions.selectedValidation) {
+    case 'Exact':
```

Comment:
should use the enums in `types/longTextField.ts` and `types/shortTextField.ts`. they seem to be duplicated, so perhaps we can cut out one of them.

Reply:
ok